### ISSUE TITLE
(aws_appsync): Unable to reference external SourceApis in Defintions.fromSourceApis()

### ISSUE THREAD

**Issue by richardsonmarkj** [aws/aws-cdk#27098]
### Describe the bug

I believe this is simply a typo in the creation of the reference type `SourceApi`. The issue is that `SourceApi.sourceApi `is of a `GraphqlApi` type. This requires that the source API be created in the stack. This should instead be a `IGraphqlApi` type so that the results of a lookup using `GraphqlApi.fromGraphqlApiAttributes`

### Expected Behavior

The following Typescript code compiles:

``` typescript
import * as cdk from "aws-cdk-lib"
import {Definition, GraphqlApi, MergeType} from "aws-cdk-lib/aws-appsync"
import {Construct} from "constructs"

export class BadStack extends cdk.Stack {
	constructor(scope:Construct, id:string, props?: cdk.StackProps) {
		super(scope, id, props)
		
		const apiId = "API_ID"
		const apiArn = "API_ARN"
		const api = new GraphqlApi(this, 'merged-api', {
			name: 'MergedAPI',
			definition: Definition.fromSourceApis({
				sourceApis: [
					{
						sourceApi: GraphqlApi.fromGraphqlApiAttributes(this, 'source-api', {
							graphqlApiId: apiId,
							graphqlApiArn: apiArn,
						}),
						mergeType: MergeType.AUTO_MERGE,
					}
				]
			})
		})
	}
}
```

### Current Behavior

The code fails to compile reporting the following error

```
TS2740: Type 'IGraphqlApi' is missing the following properties from type 'GraphqlApi': graphqlUrl, name, schema, modes, and 24 more.  
graphqlapi.d.ts(304, 14): The expected type comes from property 'sourceApi' which is declared here on type 'SourceApi'
```

### Reproduction Steps

``` typescript
import * as cdk from "aws-cdk-lib"
import {Definition, GraphqlApi, MergeType} from "aws-cdk-lib/aws-appsync"
import {Construct} from "constructs"

export class BadStack extends cdk.Stack {
	constructor(scope:Construct, id:string, props?: cdk.StackProps) {
		super(scope, id, props)
		
		const apiId = "API_ID"
		const apiArn = "API_ARN"
		const api = new GraphqlApi(this, 'merged-api', {
			name: 'MergedAPI',
			definition: Definition.fromSourceApis({
				sourceApis: [
					{
						sourceApi: GraphqlApi.fromGraphqlApiAttributes(this, 'source-api', {
							graphqlApiId: apiId,
							graphqlApiArn: apiArn,
						}),
						mergeType: MergeType.AUTO_MERGE,
					}
				]
			})
		})
	}
}
```

After `cdk init app --language typescript` replace the contents of `./lib/<name>-stack.ts` with the above code. The compiler should report the error
```
TS2740: Type 'IGraphqlApi' is missing the following properties from type 'GraphqlApi': graphqlUrl, name, schema, modes, and 24 more.  
graphqlapi.d.ts(304, 14): The expected type comes from property 'sourceApi' which is declared here on type 'SourceApi'
```

### Possible Solution

Change the type of `SourceApi.sourceApi` to IGraphqlApi

### CDK CLI Version

2.94.0 (build 987c329)

### Framework Version

2.95.1

### Node.js Version

v18.7.0

### OS

MacOs

### Language

Typescript

### Language Version

3.9.7

**Comment by peterwoodworth** [comment]
Yes, as far as I can tell this _should_ be able to seamlessly accept an `IGraphqlApi` - as the only values that are read that I can find are accessible on `IGraphqlApi`. Thanks for the issue

https://github.com/aws/aws-cdk/blob/ae455d822c9da2e4a1b23c548e9d220d8e95a346/packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts#L704-L730

**Comment by ndejaco2** [comment]
The PR I have in progress should partly address this as it would allow you to create the SourceApiAssociation as a standalone construct for the cases when the source api or merged api is not in the same stack as long as you have the identifier (api id for same account / ARN for cross account use case )

**Comment by scanlonp** [comment]
This was fixed by #27121. If this does not address your issue, ping me on this thread.
